modules/tinymce/src/core/main/ts/delete/MergeBlocks.ts

```diff
@@ -55,10 +55,29 @@ const nestedBlockMerge = (
 
 const sidelongBlockMerge = (rootNode: SugarElement<Node>, fromBlock: SugarElement<Element>, toBlock: SugarElement<Element>): Optional<CaretPosition> => {
   if (Empty.isEmpty(toBlock)) {
-    Remove.remove(toBlock);
     if (Empty.isEmpty(fromBlock)) {
-      PaddingBr.fillWithPaddingBr(fromBlock);
+      const inlineToBlockDescendants = Traverse.firstChild(toBlock).fold(
+        Fun.constant([]),
+        (child) => {
+          const descendants = PredicateFilter.descendants(child, ElementType.isInline);
+          return ElementType.isInline(child) ? [ child, ...descendants ] : descendants;
+        }
+      );
```

Comment:
Thinking something like this just grab the first child down the tree. Not sure if these structures are common since they would all need to be empty for this to fail. Like `<p><b></b><i></i></p>` the code you got would still get the descendants of the child so `<p><b><i></i><u></u></b></p>` would grab B, I, U while it should only grab B and I.

```suggestion
      const getInlineToBlockDescendants = (el: SugarElement<Element>) => {
         const helper = (node: SugarElement<Node>, elements: SugarElement<Element>[]) => {
            Traverse.firstChild(node).fold(
              () => elements,
              (child) => SugarNode.isElement(child) ? helper(child, elements.concat(child)) : elements
            );
         };

         return helper(el, []);
      };
      const inlineToBlockDescendants = getInlineToBlockDescendants(toBlock);
```

Reply:
Thanks, missed that. Updated the code and added the related test.